src/AppInstallerCommonCore/Public/winget/WindowsFeature.h

```diff
@@ -0,0 +1,159 @@
+            DismRestartPossible,
+            DismRestartRequired
+        } DismRestartType;
+
+#pragma pack(push, 1)
+
+        typedef struct _DismCustomProperty
+        {
+            PCWSTR Name;
+            PCWSTR Value;
+            PCWSTR Path;
+        } DismCustomProperty;
+
+        typedef struct _DismFeature
+        {
+            PCWSTR FeatureName;
+            DismPackageFeatureState State;
+        } DismFeature;
+
+        typedef struct _DismFeatureInfo
+        {
+            PCWSTR FeatureName;
+            DismPackageFeatureState FeatureState;
+            PCWSTR DisplayName;
+            PCWSTR Description;
+            DismRestartType RestartRequired;
+            DismCustomProperty* CustomProperty;
+            UINT CustomPropertyCount;
+        } DismFeatureInfo;
+
+#pragma pack(pop)
+
+#ifdef __cplusplus
+    }
+#endif
+
+#endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_DISM) */
+#pragma endregion
+
+#endif // _DISMAPI_H_
+
+
+    struct DismApiHelper
+    {
+        DismApiHelper();
+        ~DismApiHelper();
+
+        DismFeatureInfo* GetFeatureInfo(const std::string_view& name);
+        HRESULT EnableFeature(const std::string_view& name);
+        HRESULT DisableFeature(const std::string_view& name);
+
+    private:
+        using DismInitializePtr = HRESULT(WINAPI*)(int, PCWSTR, PCWSTR);
+        using DismOpenSessionPtr = HRESULT(WINAPI*)(PCWSTR, PCWSTR, PCWSTR, UINT*);
+        using DismShutdownPtr = HRESULT(WINAPI*)();
+        using DismGetFeatureInfoPtr = HRESULT(WINAPI*)(UINT, PCWSTR, PCWSTR, DismPackageIdentifier, DismFeatureInfo**);
+        using DismEnableFeaturePtr = HRESULT(WINAPI*)(UINT, PCWSTR, PCWSTR, DismPackageIdentifier, BOOL, PCWSTR*, UINT, BOOL, HANDLE, DISM_PROGRESS_CALLBACK, PVOID);
+        using DismDisableFeaturePtr = HRESULT(WINAPI*)(UINT, PCWSTR, PCWSTR, BOOL, HANDLE, DISM_PROGRESS_CALLBACK, PVOID);
+        using DismDeletePtr = HRESULT(WINAPI*)(VOID*);
+
+        typedef UINT DismSession;
+
+        wil::unique_hmodule m_module;
+        DismSession m_session = DISM_SESSION_DEFAULT;
+        DismFeatureInfo* m_featureInfo = nullptr;
+
+        DismInitializePtr m_dismInitialize = nullptr;
+        DismOpenSessionPtr m_dismOpenSession = nullptr;
+        DismGetFeatureInfoPtr m_dismGetFeatureInfo = nullptr;
+        DismEnableFeaturePtr m_dismEnableFeature = nullptr;
+        DismDisableFeaturePtr m_dismDisableFeature = nullptr;
+        DismShutdownPtr m_dismShutdown = nullptr;
+        DismDeletePtr m_dismDelete = nullptr;
+
+        void Initialize();
+        void OpenSession();
+        void Delete();
+        void Shutdown();
+    };
+
+    struct WindowsFeature
+    {
+        WindowsFeature(const std::string& name) : m_featureName{ name } {};
+
+        bool DoesExist();
+        bool IsEnabled();
+        HRESULT EnableFeature();
+        HRESULT DisableFeature();
+
+    private:
+        DismApiHelper m_dismApiHelper;

```

Comment:
There is a lot of repeated loading and probing for the functions using this pattern.  It would be much better to just have this created one time.

If you want dism to be able to unload (which seems reasonable), then I would design this differently as well.  I would then create a single `DismHelper` object in the workflow and have it create `WindowsFeature` objects for you.